tests/e2e/README.md

```diff
@@ -1,4 +1,204 @@
+    but not check what chains are currently running.
+* After each action, the state of the system is queried and compared against the expected state.
+
+## Running e2e tests
+
+If you just want to run the end-to-end tests,
+see the commands in the Makefile in the repo root.
+
+If you want to run the tests with a bit more control, see the help by running 
+```go run ./tests/e2e/... --help```
+in the repo root to see how to do that.
+
+## Defining a new test case
+
+This section explains how to define a new test case. For now, let's assume that
+all actions and state checks we want to perform already exist (see [actions.go](actions.go)
+for possible actions and [state.go](state.go) for possible state checks).
+Then what we need to do is the following:
+* Create a new test config (or decide on an existing one to reuse), see [config.go](config.go).
+The test config governs the config parameters of validators and chains that can be run in the test,
+for example we can set the genesis parameters of a chain using `ChainConfig.GenesisChanges`.
+* Define a sequence of actions and state checks to perform for our test case.
+* Add the new test case to the main file [main.go](main.go).
+    * ...in the stepChoices slice, where you need to put your step slice and the test config you want to run your steps on.
+    * ...add your test config to the testConfigs map (which is used to specify configs when calling from the CLI)
+    * ...if your test case should run by default (if you're not sure, it probably should), also add it to the predefined test cases in
+    getTestCases (which governs which test cases will be run when no specific test case is specified on the CLI)
+
+For example, a short sequence of actions and state checks could look like this:
+```
+steps := []Step{
+    Action: delegateTokensAction{
+        Chain:  ChainID("provi"),
+        From:   ValidatorID("alice"),
+        To:     ValidatorID("alice"),
+        Amount: 11000000,
+    },
+    State: State{
+        ChainID("provi"): ChainState{
+            ValPowers: &map[ValidatorID]uint{
+                ValidatorID("alice"): 511,
+                ValidatorID("bob"):   500,
+                ValidatorID("carol"): 500,
+            },
+        },
+    },
+    Action: delegateTokensAction{
+        Chain:  ChainID("provi"),
+        From:   ValidatorID("alice"),
+        To:     ValidatorID("bob"),
+        Amount: 99000000,
+    },
+    State: State{
+        ChainID("provi"): ChainState{
+            ValPowers: &map[ValidatorID]uint{
+                ValidatorID("alice"): 511,
+                ValidatorID("bob"):   599,
+                ValidatorID("carol"): 500,
+            },
+        },
+    },
+},
+```
+In this sequence, we first use the `delegateTokensAction`
+to delegate 11000000 tokens from alice to herself,
+then check that alices voting power was set to 511 (note that 11000000 tokens correspond to 11 voting power here),
```

Comment:
nit: maybe just mention this is `uatom`s? I guess that's the case right and that's why it corresponds to 11?